components/transactions/TransactionsDownloadCSV.js

```diff
@@ -119,14 +131,20 @@ const TransactionsDownloadCSV = ({ collective, client, query }) => {
 
     event.preventDefault();
 
-    const csv = await fetch(downloadUrl(), {
-      method: 'GET',
-      headers: {
-        Authorization: `Bearer ${accessToken}`,
-      },
-    }).then(response => response.text());
-
-    return exportFile('text/csv;charset=utf-8', `${collective.slug}-transactions.csv`, csv);
+    try {
+      setLoading('v2');
+      const csv = await fetch(downloadUrl(), {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }).then(response => response.text());
```

Comment:
I'm not sure that this will throw any exception in most cases.

Should we add some error handling to explicitly throw exceptions?

Reply:
> the fetch() API only rejects a promise when a “network error is encountered, although this usually means permissions issues or similar.” Basically fetch() will only reject a promise if the user is offline, or some unlikely networking error occurs, such a DNS lookup failure.

https://www.tjvantoll.com/2015/09/13/fetch-and-errors/

Reply:
Good catch :+1: I added support for that in https://github.com/opencollective/opencollective-frontend/pull/6911/commits/8d2c8677d1a7f97818e2fb6da3cf35d4467d8ded, along with some fixes to other `fetch` actions that were impacted too (Transactions > `Download receipt`).

![Peek 2021-09-23 11-57](https://user-images.githubusercontent.com/1556356/134489090-88866de9-ed9a-4849-aef3-f4abb5ebbe8e.gif)